File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        std::stack<string> st;
        for(auto elem : tokens){
            if(elem== "+" || elem=="-" || elem=="*" || elem=="/"){
                int b = stoi(st.top());
                st.pop();
                int a = stoi(st.top());
                st.pop();
                if(elem=="+") st.push(to_string(a+b));
                if(elem=="-") st.push(to_string(a-b));
                if(elem=="*") st.push(to_string(a*b));
                if(elem=="/") st.push(to_string(a/b));
            }
            else st.push(elem);
        }
        return stoi(st.top());       
        
    }
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = []
        for elem in tokens:
            if elem=="+" or elem=="-" or elem=="*" or elem=="/":
                b = st[-1]
                st.pop()
                a = st[-1]
                st.pop()
                if elem=="+": st.append(a+b)
                elif elem=="-": st.append(a-b)
                elif elem=="*": st.append(a*b)
                elif elem=="/": st.append(int(a/b))
            else: st.append(int(elem))
        
        return st[0]
        